### ISSUE TITLE
I can't create trocr python enviroment on windows 10.

### ISSUE THREAD

**Issue by doubletaro** [microsoft/unilm#702]
**Describe the bug**
I did a tutorial, but I get an error.

- Platform: Windows 10
- Python version: 3.7.13 (conda)
- PyTorch version (GPU?): 1.7.1+cu110 (GTX 1070 laptop)

I try following below command.

Installation
--------------------------------------------------------
conda create -n trocr python=3.7
conda activate trocr
git clone https://github.com/microsoft/unilm.git
cd unilm
cd trocr
pip install pybind11
pip install -r requirements.txt
pip install -v --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" 'git+https://github.com/NVIDIA/apex.git'
--------------------------------------------------------

I tried [pip install -r requirements.txt] step but error happend.
So I modified the text as follows:
The build was successful.
--------------------------------------------------------
-f https://download.pytorch.org/whl/torch_stable.html
-f https://dl.fbaipublicfiles.com/detectron2/wheels/cu110/torch1.7/index.html

torch==1.7.1+cu110
torchvision==0.8.2+cu110
torchaudio==0.7.2

timm==0.4.5
timm
natsort
tensorboard
nltk
h5py
numpy
fastwer
git+https://github.com/pytorch/fairseq.git@7751229
--------------------------------------------------------

**Comment by doubletaro** [comment]
After that, when I executed 
[pip install -v --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" git+https://github.com/NVIDIA/apex.git],
an error occurred.

In another virtual environment of this PC, I can run the GPU with pytorch 1.7.1.

Am I doing something wrong?
I would appreciate it if you could teach me!
Thank you.
--------------------------------------------------------
  RuntimeError: Error compiling objects for extension
  Error in atexit._run_exitfuncs:
  Traceback (most recent call last):
    File "C:\Users\dread\Anaconda3\envs\trocr_u\lib\site-packages\colorama\ansitowin32.py", line 59, in closed
      return stream.closed
  ValueError: underlying buffer has been detached
  error: subprocess-exited-with-error

  × Running setup.py install for apex did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: 'C:\Users\dread\Anaconda3\envs\trocr_u\python.exe' -u -c '
  exec(compile('"'"''"'"''"'"'
  # This is <pip-setuptools-caller> -- a caller that pip uses to run setup.py
  #
  # - It imports setuptools before invoking setup.py, to enable projects that directly
  #   import from `distutils.core` to work with newer packaging standards.
  # - It provides a clear error message when setuptools is not installed.
  # - It sets `sys.argv[0]` to the underlying `setup.py`, when invoking `setup.py` so
  #   setuptools doesn'"'"'t think the script is `-c`. This avoids the following warning:
  #     manifest_maker: standard file '"'"'-c'"'"' not found".
  # - It generates a shim setup.py, for handling setup.cfg-only projects.
  import os, sys, tokenize

  try:
      import setuptools
  except ImportError as error:
      print(
          "ERROR: Can not execute `setup.py` since setuptools is not available in "
          "the build environment.",
          file=sys.stderr,
      )
      sys.exit(1)

  __file__ = %r
  sys.argv[0] = __file__

  if os.path.exists(__file__):
      filename = __file__
      with tokenize.open(__file__) as f:
          setup_py_code = f.read()
  else:
      filename = "<auto-generated setuptools caller>"
      setup_py_code = "from setuptools import setup; setup()"

  exec(compile(setup_py_code, filename, "exec"))
  '"'"''"'"''"'"' % ('"'"'C:\\Users\\dread\\AppData\\Local\\Temp\\pip-req-build-7sfm51w9\\setup.py'"'"',), "<pip-setuptools-caller>", "exec"))' --cpp_ext --cuda_ext install --record 'C:\Users\dread\AppData\Local\Temp\pip-record-mv7b5kwp\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\dread\Anaconda3\envs\trocr_u\Include\apex'
  cwd: C:\Users\dread\AppData\Local\Temp\pip-req-build-7sfm51w9\
error
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> apex
--------------------------------------------------------

**Comment by liminghao1630** [comment]
This is an apex related issue, please refer to the apex repository instructions for installation or raise an issue.

**Comment by doubletaro** [comment]
Thank you for your reply, Mr. Lee.
apex and fairseq ... many many dependencies were intricately intertwined and bothered me.
But I was able to learn letters on the GPU!

In character reasoning,
I ran it using "trocr-base-handwritten.pt", but [torchvision 0.8.2 + cu110 requires torch = = 1.7.1. But there is an incompatible torch 1.11.0. ], GPU cannot be used, but the result was confirmed!

I hope it will be implemented as standard in pytorch soon.

thank you.
